### PR TITLE
Add GCP cluster type to e2e test templates

### PR DESCRIPTION
**Pull request by csrwng**

### CODE REVIEW

**Comment by csrwng** [comment]
@stevekuznetsov @smarterclayton ptal
Hopefully soon we'll have successful gcp installs. I'd like to have a test in place that we can at least kick off manually at first.

**Comment by abhinavdahiya** [comment]
should move the test container for GCP to use the variables from env in pod too https://github.com/openshift/release/blob/34b7dfff192418892457bb2f174a7295919a83c1/ci-operator/templates/openshift/installer/cluster-launch-installer-e2e.yaml#L149

**Comment by csrwng** [comment]
@smarterclayton @stevekuznetsov we need additional permissions on the GCP service account so we can create machine service accounts with the right policies:
```
level=error msg="Error: Error applying IAM policy for project \"openshift-gce-devel-ci\": Error setting IAM policy for project \"openshift-gce-devel-ci\": googleapi: Error 403: The caller does not have permission, forbidden"
level=error
level=error msg="  on ../tmp/openshift-install-722210447/iam/main.tf line 6, in resource \"google_project_iam_member\" \"worker-compute-viewer\":"
level=error msg="   6: resource \"google_project_iam_member\" \"worker-compute-viewer\" {"
level=error msg="Error: Error applying IAM policy for project \"openshift-gce-devel-ci\": Error setting IAM policy for project \"openshift-gce-devel-ci\": googleapi: Error 403: The caller does not have permission, forbidden"
level=error msg="  on ../tmp/openshift-install-722210447/iam/main.tf line 11, in resource \"google_project_iam_member\" \"worker-storage-admin\":"
level=error msg="  11: resource \"google_project_iam_member\" \"worker-storage-admin\" {"
level=error msg="Error: Error applying IAM policy for project \"openshift-gce-devel-ci\": Error setting IAM policy for project \"openshift-gce-devel-ci\": googleapi: Error 403: The caller does not have permission, forbidden"
level=error msg="  on ../tmp/openshift-install-722210447/master/main.tf line 6, in resource \"google_project_iam_member\" \"master-compute-admin\":"
level=error msg="   6: resource \"google_project_iam_member\" \"master-compute-admin\" {"
level=error msg="Error: Error applying IAM policy for project \"openshift-gce-devel-ci\": Error setting IAM policy for project \"openshift-gce-devel-ci\": googleapi: Error 403: The caller does not have permission, forbidden"
level=error msg="  on ../tmp/openshift-install-722210447/master/main.tf line 11, in resource \"google_project_iam_member\" \"master-network-admin\":"
level=error msg="  11: resource \"google_project_iam_member\" \"master-network-admin\" {"
level=error msg="Error: Error applying IAM policy for project \"openshift-gce-devel-ci\": Error setting IAM policy for project \"openshift-gce-devel-ci\": googleapi: Error 403: The caller does not have permission, forbidden"
level=error msg="  on ../tmp/openshift-install-722210447/master/main.tf line 16, in resource \"google_project_iam_member\" \"master-compute-security\":"
level=error msg="  16: resource \"google_project_iam_member\" \"master-compute-security\" {"
level=error msg="Error: Error applying IAM policy for project \"openshift-gce-devel-ci\": Error setting IAM policy for project \"openshift-gce-devel-ci\": googleapi: Error 403: The caller does not have permission, forbidden"
level=error msg="  on ../tmp/openshift-install-722210447/master/main.tf line 21, in resource \"google_project_iam_member\" \"master-storage-admin\":"
level=error msg="  21: resource \"google_project_iam_member\" \"master-storage-admin\" {"
level=error msg="Error: Error applying IAM policy for project \"openshift-gce-devel-ci\": Error setting IAM policy for project \"openshift-gce-devel-ci\": googleapi: Error 403: The caller does not have permission, forbidden"
level=error msg="  on ../tmp/openshift-install-722210447/master/main.tf line 26, in resource \"google_project_iam_member\" \"master-object-storage-admin\":"
level=error msg="  26: resource \"google_project_iam_member\" \"master-object-storage-admin\" {"
```

**Comment by abhinavdahiya** [comment]
```
        {
            "apiVersion": "config.openshift.io/v1",
            "kind": "ClusterOperator",
            "metadata": {
                "creationTimestamp": "2019-07-31T19:13:07Z",
                "generation": 1,
                "name": "cloud-credential",
                "resourceVersion": "3808",
                "selfLink": "/apis/config.openshift.io/v1/clusteroperators/cloud-credential",
                "uid": "3ae0fe64-b3c7-11e9-b18d-42010a000004"
            },
            "spec": {},
            "status": {
                "conditions": [
                    {
                        "lastTransitionTime": "2019-07-31T19:13:21Z",
                        "message": "3 of 3 credentials requests are failing to sync.",
                        "reason": "CredentialsFailing",
                        "status": "True",
                        "type": "Degraded"
                    },
                    {
                        "lastTransitionTime": "2019-07-31T19:13:21Z",
                        "message": "0 of 3 credentials requests provisioned, 3 reporting errors.",
                        "reason": "Reconciling",
                        "status": "True",
                        "type": "Progressing"
                    },
                    {
                        "lastTransitionTime": "2019-07-31T19:13:07Z",
                        "status": "True",
                        "type": "Available"
                    },
                    {
                        "lastTransitionTime": "2019-07-31T19:13:07Z",
                        "status": "True",
                        "type": "Upgradeable"
                    }
                ],
                "extension": null,
                "relatedObjects": [
                    {
                        "group": "",
                        "name": "openshift-cloud-credential-operator",
                        "resource": "namespaces"
                    }
                ],
                "versions": [
                    {
                        "name": "operator",
                        "version": "0.0.1-2019-07-31-190029"
                    }
                ]
            }
        },
```

**Comment by csrwng** [comment]
```
time="2019-07-31T19:13:18Z" level=warning msg="Detected some unallowed permissions: [iam.roles.get iam.serviceAccountKeys.create iam.serviceAccountKeys.delete iam.serviceAccounts.create iam.serviceAccounts.delete iam.serviceAccounts.get resourcemanager.projects.get resourcemanager.projects.getIamPolicy resourcemanager.projects.setIamPolicy serviceusage.services.list]" controller=secretannotator
time="2019-07-31T19:13:18Z" level=warning msg="Detected some unallowed permissions: [iam.roles.get resourcemanager.projects.get serviceusage.services.list]" controller=secretannotator
time="2019-07-31T19:13:18Z" level=warning msg="Cloud creds unable to be used for either minting or passthrough" controller=secretannotator
```

**Comment by csrwng** [comment]
Cluster creation and teardown are now functional. Can we get this merged so we can iterate on fixing failing e2e tests? For now it would only be part of the installer tests.
/cc @abhinavdahiya

File: ci-operator/jobs/openshift/installer/openshift-installer-master-presubmits.yaml [hunk 1]
```diff
@@ -586,6 +586,72 @@ presubmits:
...
+    branches:
+    - master
+    context: ci/prow/e2e-gcp
+    decorate: true
+    decoration_config:
+      skip_cloning: true
+    labels:
+      ci-operator.openshift.io/prowgen-controlled: "true"
+    name: pull-ci-openshift-installer-master-e2e-gcp
+    rerun_command: /test e2e-gcp
```
Comment (abhinavdahiya) [inline ci-operator/jobs/openshift/installer/openshift-installer-master-presubmits.yaml]: needs `optional: true`

File: ci-operator/jobs/openshift/installer/openshift-installer-master-presubmits.yaml [hunk 2]
```diff
@@ -586,6 +586,72 @@ presubmits:
         secret:
           secretName: sentry-dsn
     trigger: (?m)^/test( | .* )e2e-etcd-quorum-loss,?($|\s.*)
+  - agent: kubernetes
+    always_run: true
```
Comment (abhinavdahiya) [inline ci-operator/jobs/openshift/installer/openshift-installer-master-presubmits.yaml]: I think we should start with `false`

**Pull request merged**